components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1315,64 +1312,30 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
             //
             // If that, source peer still need to decide whether to destroy itself. When the target
             // peer has already moved on, source peer can destroy itself.
-            if epoch.get_version() > merge_target.get_region_epoch().get_version() {
+            if util::is_epoch_stale(merge_target.get_region_epoch(), r.get_region_epoch()) {
                 return Ok(true);
             }
             return Ok(false);
         }
+        drop(meta);
 
-        // Check whether target peer is set to tombstone already.
```

Comment:
Can they be kept in case PD is not working as expected?

Reply:
I don't think so. The query to PD can only handle some cases not all of them. I think it has very little possibility that PD does not ensure all target peer exists because I have already checked them. Also, if PD really does not ensure it, the consequence is not serious and we can remove the region manually.